core/src/test/java/org/pentaho/di/connections/vfs/provider/BaseVFSConnectionProviderTest.java

```diff
@@ -0,0 +1,85 @@
+
+package org.pentaho.di.connections.vfs.provider;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Answers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.pentaho.di.connections.utils.VFSConnectionTestOptions;
+import org.pentaho.di.connections.vfs.BaseVFSConnectionDetails;
+import org.pentaho.di.connections.vfs.BaseVFSConnectionProvider;
+import org.pentaho.di.connections.vfs.VFSConnectionDetails;
+import org.pentaho.di.core.exception.KettleException;
+
+import static org.mockito.Mockito.when;
+
+public class BaseVFSConnectionProviderTest {
+  @Mock
+  BaseVFSConnectionDetails baseVFSConnectionDetails;
```

Comment:
I have not looked at the test code, I have questions about the source code implementation. 
- https://github.com/pentaho/pentaho-kettle/pull/9305/files#r1589424773
- https://github.com/pentaho/pentaho-kettle/pull/9305/files#r1589437196

However, this is not how you use mocking libraries. Never mock out the class under test, in this case `BaseVFSConnectionProvider.java`.

Please look at https://github.com/pentaho/pentaho-scheduler-plugin-ee/blob/f4540fedef10872d2df78045bbc47f00574503d8/core/src/test/java/com/pentaho/vfs/connections/service/ConnectionManagerServiceTest.java for mocking guidance.

Reply:
I think my concerns have largely been address by: https://github.com/pentaho/pentaho-kettle/pull/9305/files#r1592967128